doctest/doctest.h

```diff
@@ -1162,7 +1174,11 @@ DOCTEST_CLANG_SUPPRESS_WARNING_WITH_PUSH("-Wunused-comparison")
 // If not it doesn't find the operator or if the operator at global scope is defined after
 // this template, the template won't be instantiated due to SFINAE. Once the template is not
 // instantiated it can look for global operator using normal conversions.
+#if !DOCTEST_ICC
 #define SFINAE_OP(ret,op) decltype((void)(doctest::detail::declval<L>() op doctest::detail::declval<R>()),ret{})
+#else
+#define SFINAE_OP(ret,op) ret
+#endif
```

Comment:
This actually breaks for me using ICC (19.2). Simply removing the `#if` fixes it.

Reply:
@isaevil would it work without these last few lines of changes?

Reply:
@onqtam @Saalvage No, it doesn't work for me. I am getting this kind of error:
```
doctest.h(853): error: too many recursive substitutions of function template signatures
      template<typename T> auto declval() DOCTEST_NOEXCEPT -> decltype(declval<T>(0)) ;
```

Reply:
So this will need another switch based on whether it's running on Windows